pkg/osquery/extension.go

```diff
@@ -403,19 +400,26 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 		return "", true, fmt.Errorf("generating UUID: %w", err)
 	}
 
-	// We've seen this fail, so add some retry logic.
+	// We used to see the enrollment details fail, but now that we're running as an exec,
+	// it seems less likely. Try a couple times, but backoff fast.
 	var enrollDetails service.EnrollmentDetails
-	if err := backoff.WaitFor(func() error {
-		enrollDetails, err = getEnrollDetails(e.osqueryClient)
-		return err
-	}, 2*time.Minute, 10*time.Second); err != nil {
-		if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_REQUIRED") == "true" {
-			return "", true, fmt.Errorf("query enrollment details: %w", err)
-		}
+	if e.knapsack.OsquerydPath() == "" {
+		level.Info(logger).Log("msg", "Cannot get additional enrollment details without an osqueryd path. This is probably CI")
+	} else {
+		if err := backoff.WaitFor(func() error {
+			enrollDetails, err = getEnrollDetails(e.knapsack.OsquerydPath())
+			if err != nil {
+				level.Debug(logger).Log("msg", "getEnrollDetails failed in backoff", "err", err)
+			}
+			return err
+		}, 5*time.Second, 5*time.Second); err != nil {
```

Comment:
This will only run once -- we should make the timeout larger than the interval. Maybe --

```suggestion
		}, 30*time.Second, 5*time.Second); err != nil {
```